lib/beacon/broadcast.go

```diff
@@ -149,35 +90,25 @@ func (w *broadcastWriter) serve(stop chan struct{}) error {
 			if err != nil {
 				// Some other error that we don't expect. Debug and continue.
 				l.Debugln(err)
-				w.SetError(err)
```

Comment:
This changes behaviour?

Reply:
Or I guess this compliments the success condition change below.

Reply:
Exactly: Before we set a transient error for every loop iteration. However we didn't do it consistently: If a new error comes up, that gets set, but if something succeeds, the error doesn't get set. I don't see the value of exposing this intermediate errors. Now we just expose an error if none of the tries succeed (inverse condition to removing the error again previously).